pkg/services/sqldb/bind.go

```diff
@@ -168,6 +168,10 @@ func (d *dbOnlyManager) Bind(
 			"error casting instance.Details as *mssqlDBOnlyInstanceDetails",
 		)
 	}
+	//Parent should be set by the framework, but return an error if it is not set.
```

Comment:
I added these checks because when I first added the life cycle test (but before I was implementing setup), there was no parent instance and there was no check here (framework SHOULD set it before this gets invoked) so the result was a panic. This seemed safer from a runtime perspective, but I can remove this.

Reply:
A justifiable degree of defensive coding is rarely a bad thing.